Source/ASDisplayNode+Yoga.mm

```diff
@@ -31,8 +32,17 @@ - (ASSizeRange)_locked_constrainedSizeForLayoutPass;
 
 @implementation ASDisplayNode (Yoga)
 
+- (ASDisplayNode *)yogaRoot {
+  ASDisplayNode *yogaRoot = self;
+  while(yogaRoot.yogaParent) {
+    yogaRoot = yogaRoot.yogaParent;
+  }
+  return yogaRoot;
+}
+
 - (void)setYogaChildren:(NSArray *)yogaChildren
 {
+  ASLockScope(self.yogaRoot);
```

Comment:
Is this safe? Can't the yogaRoot be changing at this time? I'm pretty sure we also need to lock self, before accessing this and also hold that lock for the duration of the method for it to be re-entrant safe.